pkg/crypto/crypto.go

```diff
@@ -828,15 +828,32 @@ func (ca *CA) MakeServerCertForDuration(hostnames sets.String, lifetime time.Dur
 }
 
 func (ca *CA) EnsureClientCertificate(certFile, keyFile string, u user.Info, expireDays int) (*TLSCertificateConfig, bool, error) {
-	certConfig, err := GetTLSCertificateConfig(certFile, keyFile)
+	certConfig, err := ca.GetClientCertificate(certFile, keyFile, u)
 	if err != nil {
 		certConfig, err = ca.MakeClientCertificate(certFile, keyFile, u, expireDays)
 		return certConfig, true, err // true indicates we wrote the files.
 	}
-
 	return certConfig, false, nil
 }
 
+func (ca *CA) GetClientCertificate(certFile, keyFile string, u user.Info) (*TLSCertificateConfig, error) {
+	certConfig, err := GetTLSCertificateConfig(certFile, keyFile)
+	if err != nil {
+		return nil, err
+	}
+
+	// Validate if the existing certificate subject has changed, if it has we need to regenerate the certificate
+	// The Subject field is converted to String (RFC 2253 Distinguished Names syntax) because disk
+	// encoding does not match 100% to struct encoding and DeepCompare fails when it should match
+	cert := certConfig.Certs[0]
+	if cert.Subject.String() != userToSubject(u).String() {
```

Comment:
After reading the internets I'd rather if we just compared the three aspects that we care about than rely on the implementation of `String().`

I think the best way would be to:
1. compare `CommonName` and `SerialNumber` of the subjects
2. order `Organization` fields of both and compare them

Reply:
Ack I'll update the pr

Reply:
Done